module/zfs/dsl_scan.c

```diff
@@ -5314,4 +5352,7 @@ ZFS_MODULE_PARAM(zfs, zfs_, resilver_defer_percent, UINT, ZMOD_RW,
 
 ZFS_MODULE_PARAM(zfs, zfs_, scrub_error_blocks_per_txg, UINT, ZMOD_RW,
 	"Error blocks to be scrubbed in one txg");
+
+ZFS_MODULE_PARAM(zfs, zfs_, scrub_recent_txgs, UINT, ZMOD_RW,
+	"The number of TXGs should be scrubbed while scrubbing recent data");
```

Comment:
Sure it was trivial to implement, but I find this interface ugly and not user-friendly.  User has no easy way to know how many TXGs he needs, and setting it via module parameter does not allow to use it flexibly for several pools.  I'd expect some way of specification it on command line, and in some more user-friendly way rather than number of TXGs.

Reply:
This is one of the options that I/we think might be useful for the user. It just defines how much is the "latest", and it's an arbitrary number I agree. We have follow-up changes to be able to scrub by date range not txgs.

Reply:
We can also split it to seperate PR if you find it controversial to not block other changes :)

Reply:
To be fair, I don't like that scrub during import also, but primarily due the time it takes during import.  But it is just a (relatively) quick sanity check that pool was not badly corrupted by something catastrophic before export to allow user do some recovery by rollback while it might still be possible.

When user explicitly runs the scrub though, he likely expects some sort of real-life guaranties, like that the files he has just written got to stable storage correctly.  And in that sense 256 TXGs can mean anything from couple minutes to infinity, which is not what I associate with guaranties.  As I have told above, I understand that it was just easy to implement it how it is now, but I think even if we allow range specification in terms of TXGs, it should better be done via command line parameter, if user can get TXG number out of somewhere, like snapshot's `createtxg`, but really better in take some real terms of snapshot, bookmark, file name etc.

I worry that it may become one of those temporary oddities that stick forever.  I'd prefer something better designed for long term, especially if you are working on it already.

Reply:
On a level of ideas, would the `last_scrubbed_txg` property be writable, you could set it to a wanted value (negative value could mean last number of TXGs as you want it) and then run scrub from that "last" position.

Reply:
Ok, I have retracted the change about the recent txgs :)